packages/api-contract/src/types.ts

```diff
@@ -55,8 +55,8 @@ export interface InterfaceContractCalls {
 
 export interface ContractCallOutcome {
   debugMessage: Text;
-  gasConsumed: u64;
-  gasRequired: u64;
+  gasConsumed: WeightV2;
```

Comment:
The next one is correct - the Weight type would be detected on-chain, so using “Weight” will result in the correct on-chain type (which could be v1, v1.5 or v2)

this is important using the detect here since different chains would have different requirements (based on their upgrade cycle)

So the best approach is to change all the different definitions, where weight is involved, to “Weight”.

Reply:
Changed to `Weight` and force pushed

Reply:
Should this line https://github.com/polkadot-js/api/blob/master/packages/api-contract/src/types.ts#L76 change to `Weight` too?

Reply:
https://github.com/polkadot-js/api/blob/master/packages/api-contract/src/types.ts#L87
Should I need to add `proof_size` for `v2Weight` ?

Reply:
The interface (first question) is fine. It detects what is on-chain and then passes the correct construct through, either for v1, v1.5 or v2

Reply:
For the second - no. That is an internal interface that just manages the shape for compat reasons.

Reply:
I changed https://github.com/polkadot-js/api/blob/master/packages/api-contract/src/types.ts#L76 to `Weight` and don't touch `AllWeight`